sound/soc/sof/intel/hda.c

```diff
@@ -1190,7 +1192,22 @@ void hda_machine_select(struct snd_sof_dev *sdev)
 		dev_warn(sdev->dev, "warning: No matching ASoC machine driver found\n");
 }
 
+int hda_pci_intel_probe(struct pci_dev *pci, const struct pci_device_id *pci_id)
+{
+	int ret;
+
+	ret = snd_intel_dsp_driver_probe(pci);
+	if (ret != SND_INTEL_DSP_DRIVER_ANY && ret != SND_INTEL_DSP_DRIVER_SOF) {
+		dev_dbg(&pci->dev, "SOF PCI driver not selected, aborting probe\n");
+		return -ENODEV;
+	}
+
+	return sof_pci_probe(pci, pci_id);
+}
+EXPORT_SYMBOL_NS(hda_pci_intel_probe, SND_SOC_SOF_INTEL_HDA_COMMON);
+
 MODULE_LICENSE("Dual BSD/GPL");
+MODULE_IMPORT_NS(SND_SOC_SOF_PCI_DEV);
```

Comment:
It seems we now have following dependency 
```
snd_sof_pci_intel_apl    16384  0
snd_sof_intel_hda_common    86016  1 snd_sof_pci_intel_apl
snd_sof_pci            20480  2 snd_sof_intel_hda_common,snd_sof_pci_intel_apl
```
@plbossart We may need a PR for PCI dependency orders

Reply:
I don't understand the comment, what is wrong here?